visualisation/2dCompImage.cpp

```diff
@@ -0,0 +1,245 @@
+ *
+ * This file is part of the DGtal library/DGtalTools Project.
+ */
+
+///////////////////////////////////////////////////////////////////////////////
+#include "DGtal/base/Common.h"
+#include "DGtal/helpers/StdDefs.h"
+#include <DGtal/io/readers/GenericReader.h>
+#include <DGtal/io/writers/GenericWriter.h>
+
+
+#include <DGtal/images/Image.h>
+#include <DGtal/images/ImageContainerBySTLVector.h>
+#include <DGtal/images/imagesSetsUtils/SetFromImage.h>
+#include <DGtal/io/writers/PPMWriter.h>
+#include <DGtal/math/Statistic.h>
+#include "DGtal/io/colormaps/GradientColorMap.h"
+#include <DGtal/math/Statistic.h>
+
+#include <boost/program_options/options_description.hpp>
+#include <boost/program_options/parsers.hpp>
+#include <boost/program_options/variables_map.hpp>
+///////////////////////////////////////////////////////////////////////////////
+using namespace std;
+using namespace DGtal;
+///////////////////////////////////////////////////////////////////////////////
+
+typedef ImageContainerBySTLVector < Z2i::Domain,  unsigned char > Image2D;
+typedef ImageContainerBySTLVector < Z2i::Domain,  unsigned int > Image2DErr;
+typedef GradientColorMap<unsigned int, CMAP_JET, 1 > JetMap;
+
+namespace po = boost::program_options;
+
+
+/**
+ @page Doc2dCompImage 2dCompImage
+ 
+ @brief Computes and displays image comparisons (squared and absolute differences). 
+
+
+ @b Usage:  2dCompImage --imageA <imageA>.pgm --imageB <imageB>.pgm --imageError <name> 
+
+ @b Allowed @b options @b are :
+ 
+ @code
+  -h [ --help ]                    display this message
+  -a [ --imageA ] arg              Input filename of image A.
+  -b [ --imageB ] arg              Input filename of image B.
+  -e [ --imageError ] arg          Output error image basename (will generate 
+                                   two images <basename>MSE.ppm and 
+                                   <basename>MAE.ppm).
+  -S [ --fixMaxColorValueMSE ] arg Fix the maximal color value for the scale 
+                                   display of MSE (else the scale is set the 
+                                   maximal MSE value).
+  -A [ --fixMaxColorValueMAE ] arg Fix the maximal color value for the scale 
+                                   display of MAE (else the scale is set from 
+                                   the maximal MAE value).
+ @endcode
+
+ @b Example: 
+ Typical use example:
+ @code
+       2dCompImage -a imageA.pgm -b imageB.pgm -e errorImage -S 100 
+ @endcode
+
+You should obtain such a visualisation:
+ @image html res2dCompImage.png "resulting visualisation of absolute error between two images."
+
+ @see
+ @ref 2dCompImage.cpp
+
+
+ */
+
+
+
+
+/**
+ * Compute statistics on the mean absolute difference between pixel values of image A and B. 
+ **/
+Statistic< int>
+getMAEstats(const Image2D & imageA, const Image2D &imageB, Image2DErr &imageMAE)
+{
+  Statistic< int> stat(false);
+  for(Image2D::Domain::ConstIterator it = imageA.domain().begin();
```

Comment:
or more simply: 

`for(auto point: imageA.domain())`

Reply:
;)

Reply:
yes thanks C++11 ;)